cmd/minikube/cmd/start_test.go

```diff
@@ -70,6 +70,50 @@ func TestGetKuberneterVersion(t *testing.T) {
 	}
 }
 
+func TestMirrorCountry(t *testing.T) {
+	// Set default disk size value in lieu of flag init
+	viper.SetDefault(humanReadableDiskSize, defaultDiskSize)
+
+	k8sVersion := constants.DefaultKubernetesVersion
+	var tests = []struct {
+		description     string
+		k8sVersion      string
+		imageRepository string
+		mirrorCountry   string
+		cfg             *cfg.ClusterConfig
+	}{
+		{
+			description:     "image-repository none, image-mirror-country none",
+			imageRepository: "",
+			mirrorCountry:   "",
+		},
+		{
+			description:     "image-repository auto, image-mirror-country none",
+			imageRepository: "auto",
+			mirrorCountry:   "",
+		},
+		{
+			description:     "image-repository auto, image-mirror-country china",
+			imageRepository: "auto",
+			mirrorCountry:   "cn",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			cmd := &cobra.Command{}
+			viper.SetDefault(imageRepository, test.imageRepository)
```

Comment:
Calling viper.Set in unit test has un-wanted side effects, such as setting the user 's settings and also can not run in parallel safely

Could we break down generateCfgFromFlags func so we can test only the mirror func without need to pass cobra ?

Reply:
No idea, think I only copied what the test above was doing or something

```
                        viper.SetDefault(kubernetesVersion, test.paramVersion)
```

Reply:
I think we can refactor our tests not to use viper as much as possible, we could parameterize the funcs into small independent testable units. what do u think? maybe good opprotunity to refactor this part of the code?